cli/flox/src/main.rs

```diff
@@ -30,8 +32,54 @@ async fn run(args: FloxArgs) -> Result<()> {
     Ok(())
 }
 
-#[tokio::main]
-async fn main() -> ExitCode {
+fn main() -> ExitCode {
+    let sentry_dns = std::env::var("SENTRY_DSN");
+    let _sentry;
+
+    if sentry_dns.is_ok() {
+        _sentry = sentry::init((sentry_dns.unwrap(), sentry::ClientOptions {
+            // https://docs.sentry.io/platforms/rust/configuration/releases/
+            // TODO: should we maybe just use commit hash
+            release: sentry::release_name!(),
+
+            // https://docs.sentry.io/platforms/rust/configuration/environments/
+            // TODO: need to set this to respective channel: nightly, ...
+            // eg. environment: std::env::var("SENTRY_ENV").unwrap_or_default("development").into(),
+            environment: Some("development".into()),
+
+            // certain personally identifiable information (PII) are added
+            send_default_pii: true,
+
+            // Enable debug mode when needed
+            debug: true,
+
+            // To set a uniform sample rate
+            // https://docs.sentry.io/platforms/rust/performance/
+            traces_sample_rate: 1.0,
+
+            ..Default::default()
+        }));
+    }
+
+    // TODO: configure user
+    //sentry::configure_scope(|scope| {
+    //    scope.set_user(Some(sentry::User {
+    //        email: Some("jane.doe@example.com".to_owned()),
+    //        ..Default::default()
+    //    }));
+    //});
+
+    let exit_code = tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(async { amain().await });
```

Comment:
Just FYI I think you can just do
```
.block_on(amain());
```
since `amain` returns a Future

Reply:
I tried but it was complaining about something so I just followed their tutorial to the letter.

I hope you will find a better name for `amain` :)

Reply:
`async_main` is pretty common :)